server/pulp/server/webservices/views/repositories.py

```diff
@@ -0,0 +1,1214 @@
+        :param repo_id: id of the repository
+        :type  repo_id: str
+
+        :return: Response containing a list of dicts, one for each sync event
+        :rtype : django.http.HttpResponse
+        :raises pulp_exceptions.InvalidValue: if limit is not an integer
+        """
+
+        sort = request.GET.get(constants.REPO_HISTORY_FILTER_SORT)
+        start_date = request.GET.get(constants.REPO_HISTORY_FILTER_START_DATE)
+        end_date = request.GET.get(constants.REPO_HISTORY_FILTER_END_DATE)
+        limit = request.GET.get(constants.REPO_HISTORY_FILTER_LIMIT)
+        if limit:
+            try:
+                limit = int(limit)
+            except ValueError:
+                raise pulp_exceptions.InvalidValue([constants.REPO_HISTORY_FILTER_LIMIT])
+        if not sort:
+            sort = constants.SORT_DESCENDING
+
+        sync_manager = manager_factory.repo_sync_manager()
+        # Error checking is done on these options in the sync manager before the database is queried
+        entries = sync_manager.sync_history(repo_id, limit=limit, sort=sort, start_date=start_date,
+                                            end_date=end_date)
+        return generate_json_response_with_pulp_encoder(entries)
+
+
+class RepoPublishHistory(View):
+    """
+    View for publish history of a specified repository.
+    """
+
+    @auth_required(authorization.READ)
+    def get(self, request, repo_id, distributor_id):
+        """
+        Retrieve publish history for a specified distributor.
+
+        :param request: WSGI request object
+        :type  request: django.core.handlers.wsgi.WSGIRequest
+        :param repo_id: id of the repository
+        :type  repo_id: str
+        :param repo_id: retrieve the publish history of this distributor
+        :type  repo_id: str
+
+        :return: Response containing a list of dicts, one for each publish event
+        :rtype : django.http.HttpResponse
+        """
+
+        sort = request.GET.get(constants.REPO_HISTORY_FILTER_SORT)
+        start_date = request.GET.get(constants.REPO_HISTORY_FILTER_START_DATE)
+        end_date = request.GET.get(constants.REPO_HISTORY_FILTER_END_DATE)
+        limit = request.GET.get(constants.REPO_HISTORY_FILTER_LIMIT)
+        if limit:
+            try:
+                limit = int(limit)
+            except ValueError:
+                raise pulp_exceptions.InvalidValue([constants.REPO_HISTORY_FILTER_LIMIT])
+        if not sort:
+            sort = constants.SORT_DESCENDING
+
+        publish_manager = manager_factory.repo_publish_manager()
+        entries = publish_manager.publish_history(repo_id, distributor_id, limit=limit, sort=sort,
+                                                  start_date=start_date, end_date=end_date)
+        return generate_json_response_with_pulp_encoder(entries)
+
+
+class RepoSync(View):
+    """
+    View for syncing a repository.
+    """
+
+    @auth_required(authorization.EXECUTE)
+    @json_body_allow_empty
```

Comment:
would you change the docs and specify that the 'override_config' is optional param?
http://pulp.readthedocs.org/en/latest/dev-guide/integration/rest-api/repo/sync.html#sync-a-repository